beancount/core/getters.py

```diff
@@ -32,7 +34,7 @@
     from beancount.core.data import Note
     from beancount.core.data import Pad
 
-    AccountsUseMap = tuple[dict[Account, datetime.date], dict[Account, datetime.date]]
+    AccountsUseMap = Tuple[Dict[Account, datetime.date], Dict[Account, datetime.date]]
```

Comment:
Why not keep the new modern types?

Reply:
I'll assume there's a mypy related reason; I'll go ahead and merge this, we can separately modernize all the type annotations after (there are plenty others).

Reply:
Let's bump it up to 3.9 and require modern type annotations.
(In general I feel in branch `master` it's fine for us to require more modern Python. 
Even 3.9 is old by now, if 3.12 had something we care about I'd bump the req.)

Reply:
Oh I wonder if gemini can modernize all the type annotations for me... I'm on an Aider kick...

Reply:
AFAIK generics in standard type annotations became possible in 3.9, so you should be alright:
https://peps.python.org/pep-0585/